NullPointer while starting a component

Crash occurs at:

```
org.mozilla.mozstumbler.client.navdrawer.MainDrawerActivity.setStartStopMenuItemState(MainDrawerActivity.java:140)
```

https://github.com/mozilla/MozStumbler/blob/v0.40.1.0/android/src/main/java/org/mozilla/mozstumbler/client/navdrawer/MainDrawerActivity.java#L140

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{org.mozilla.mozstumbler/org.mozilla.mozstumbler.client.navdrawer.MainDrawerActivity}: java.lang.NullPointerException
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2114)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2139)
at android.app.ActivityThread.access$700(ActivityThread.java:143)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1241)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loop(Looper.java:137)
at android.app.ActivityThread.main(ActivityThread.java:4960)
at java.lang.reflect.Method.invokeNative(Native Method)
at java.lang.reflect.Method.invoke(Method.java:511)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1038)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:805)
at dalvik.system.NativeStart.main(Native Method)
Caused by: java.lang.NullPointerException
at org.mozilla.mozstumbler.client.navdrawer.MainDrawerActivity.setStartStopMenuItemState(MainDrawerActivity.java:140)
at org.mozilla.mozstumbler.client.navdrawer.MainDrawerActivity.onCreateOptionsMenu(MainDrawerActivity.java:132)
at android.app.Activity.onCreatePanelMenu(Activity.java:2554)
at android.support.v4.app.FragmentActivity.onCreatePanelMenu(FragmentActivity.java:224)
at android.support.v7.app.ActionBarActivity.superOnCreatePanelMenu(ActionBarActivity.java:232)
at android.support.v7.app.ActionBarActivityDelegateICS.onCreatePanelMenu(ActionBarActivityDelegateICS.java:146)
at android.support.v7.app.ActionBarActivity.onCreatePanelMenu(ActionBarActivity.java:199)
at android.support.v7.app.ActionBarActivityDelegateICS$WindowCallbackWrapper.onCreatePanelMenu(ActionBarActivityDelegateICS.java:293)
at com.android.internal.policy.impl.PhoneWindow.preparePanel(PhoneWindow.java:455)
at com.android.internal.policy.impl.PhoneWindow.invalidatePanelMenu(PhoneWindow.java:835)
at com.android.internal.policy.impl.PhoneWindow.restorePanelState(PhoneWindow.java:1910)
at com.android.internal.policy.impl.PhoneWindow.restoreHierarchyState(PhoneWindow.java:1860)
at android.app.Activity.onRestoreInstanceState(Activity.java:973)
at android.app.Activity.performRestoreInstanceState(Activity.java:945)
at android.app.Instrumentation.callActivityOnRestoreInstanceState(Instrumentation.java:1145)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2092)
... 11 more
java.lang.NullPointerException
at org.mozilla.mozstumbler.client.navdrawer.MainDrawerActivity.setStartStopMenuItemState(MainDrawerActivity.java:140)
at org.mozilla.mozstumbler.client.navdrawer.MainDrawerActivity.onCreateOptionsMenu(MainDrawerActivity.java:132)
at android.app.Activity.onCreatePanelMenu(Activity.java:2554)
at android.support.v4.app.FragmentActivity.onCreatePanelMenu(FragmentActivity.java:224)
at android.support.v7.app.ActionBarActivity.superOnCreatePanelMenu(ActionBarActivity.java:232)
at android.support.v7.app.ActionBarActivityDelegateICS.onCreatePanelMenu(ActionBarActivityDelegateICS.java:146)
at android.support.v7.app.ActionBarActivity.onCreatePanelMenu(ActionBarActivity.java:199)
at android.support.v7.app.ActionBarActivityDelegateICS$WindowCallbackWrapper.onCreatePanelMenu(ActionBarActivityDelegateICS.java:293)
at com.android.internal.policy.impl.PhoneWindow.preparePanel(PhoneWindow.java:455)
at com.android.internal.policy.impl.PhoneWindow.invalidatePanelMenu(PhoneWindow.java:835)
at com.android.internal.policy.impl.PhoneWindow.restorePanelState(PhoneWindow.java:1910)
at com.android.internal.policy.impl.PhoneWindow.restoreHierarchyState(PhoneWindow.java:1860)
at android.app.Activity.onRestoreInstanceState(Activity.java:973)
at android.app.Activity.performRestoreInstanceState(Activity.java:945)
at android.app.Instrumentation.callActivityOnRestoreInstanceState(Instrumentation.java:1145)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2092)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2139)
at android.app.ActivityThread.access$700(ActivityThread.java:143)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1241)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loop(Looper.java:137)
at android.app.ActivityThread.main(ActivityThread.java:4960)
at java.lang.reflect.Method.invokeNative(Native Method)
at java.lang.reflect.Method.invoke(Method.java:511)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1038)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:805)
at dalvik.system.NativeStart.main(Native Method)
```

--- a/android/src/main/java/org/mozilla/mozstumbler/client/navdrawer/MainDrawerActivity.java
+++ b/android/src/main/java/org/mozilla/mozstumbler/client/navdrawer/MainDrawerActivity.java
@@ -4,6 +4,7 @@
 
 package org.mozilla.mozstumbler.client.navdrawer;
 
+import android.app.Service;
 import android.content.Intent;
 import android.content.res.Configuration;
 import android.os.Build;
@@ -137,9 +138,20 @@ public class MainDrawerActivity
         if (mMenuItemStartStop == null) {
             return;
         }
-        boolean isScanning = ((MainApp) getApplication()).getService().isScanning();
+
+        MainApp app = (MainApp) getApplication();
+        if (app == null) {
+            return;
+        }
+
+        ClientStumblerService svc = app.getService();
+        if (svc == null) {
+            return;
+        }
+        boolean isScanning = svc.isScanning();
+
         if (Build.VERSION.SDK_INT >= 11) {
-            Switch s = (Switch)mMenuItemStartStop.getActionView();
+            Switch s = (Switch) mMenuItemStartStop.getActionView();
             if (isScanning) {
                 s.setChecked(true);
             } else {

